mlflow/autogen/autogen_logger.py

```diff
@@ -0,0 +1,262 @@
+import logging
+import time
+import uuid
+from dataclasses import dataclass, field
+from datetime import datetime
+from typing import Any, Callable, Dict, List, Optional, Union
+
+from autogen import Agent, ConversableAgent
+from autogen.logger.base_logger import BaseLogger
+from openai.types.chat import ChatCompletion
+
+from mlflow import MlflowClient
+from mlflow.entities.span import NoOpSpan, Span, SpanType
+from mlflow.entities.span_event import SpanEvent
+from mlflow.entities.span_status import SpanStatus, SpanStatusCode
+from mlflow.tracing.utils import capture_function_input_args
+
+# For GroupChat, a single "received_message" events are passed around multiple
+# internal layers and thus too verbose if we show them all. Therefore we ignore
+# some of the message senders listed below.
+_EXCLUDED_MESSAGE_SENDERS = ["chat_manager", "checking_agent"]
```

Comment:
Are these only providing status updates (i.e. "I am sending this message to agent1")? Or is it replicated reception confirmation from each agent -> manager / checking agent for decision determination?

Are these messages internal-only, since the stdout reports from configured agents don't seem to have these reported out?

Reply:
The `chat_manager` is merely a facade passes the message from one agent to the next speaker agent, so it does not add any information. https://github.com/microsoft/autogen/blob/2a436c90e6e55bd3af07abbcd8ea7da1870134ec/autogen/agentchat/groupchat.py#L1100

The `checking_agent` is a bit arguable. It basically checks if the next speaker name decided by the spearker selection agent is valid or not. For example, sometimes users can define a set of valid transition between agents and block other patterns. https://github.com/microsoft/autogen/blob/2a436c90e6e55bd3af07abbcd8ea7da1870134ec/autogen/agentchat/groupchat.py#L610-L619

While this does not tweak the messages and context passed around, it may be useful for users to see why a certain transition did not happen, with some extra verbosity. Thus on second thought, I'm happy to remove the second one from the exclusion list to reflecte the native behavior on traces as much as possible WDYT?

Reply:
It feels a bit like a pure debugger feature. I think leaving it out to reduce the span 'noise' is a wise decision. We can always easily re-enable the functionality with a one-line code change in the future.